File: scratch/audit_all_elements.py
import os
import re

root_dir = r"c:\Users\vc\Documents\nls-blog-hame\nls-blog-hame"

html_files = []
for dirpath, dirnames, filenames in os.walk(root_dir):
    if '.git' in dirpath or 'brain' in dirpath or 'scratch' in dirpath:
        continue
    for f in filenames:
        if f.endswith('.html'):
            html_files.append(os.path.join(dirpath, f))

print(f"Auditing {len(html_files)} HTML files for white text / white logo on light background...")

issues = []

for file_path in html_files:
    rel_path = os.path.relpath(file_path, root_dir)
    with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
        lines = f.readlines()
        
    for idx, line in enumerate(lines):
        line_num = idx + 1
        line_clean = line.strip()
        
        # 1. Direct text-white with light background classes on same line
        if 'text-white' in line_clean and 'dark:text-white' not in line_clean:
            # check if line has light bg without dark: or solid color
            has_light_bg = any(k in line_clean for k in ['bg-white', 'bg-slate-50', 'bg-slate-100', 'bg-gray-50', 'bg-surface-bright', 'bg-surface-container-lowest'])
            has_solid_bg = any(k in line_clean for k in ['bg-sky-', 'bg-blue-', 'bg-primary', 'bg-indigo-', 'bg-emerald-', 'bg-rose-', 'bg-amber-', 'bg-slate-800', 'bg-slate-900', 'bg-black', 'bg-[#FF8A00]', 'bg-[#0284c7]', 'bg-[#131D38]', 'from-', 'bg-gradient', 'bg-secondary'])
            
            if has_light_bg and not has_solid_bg:
                issues.append((rel_path, line_num, "TEXT_WHITE_ON_LIGHT", line_clean))
                
        # 2. White logo/icon on light background
        if ('logo' in line_clean.lower() or 'icon' in line_clean.lower()) and ('text-white' in line_clean or 'stroke-white' in line_clean or 'fill-white' in line_clean):
            if 'bg-white' in line_clean and 'dark:' not in line_clean:
                issues.append((rel_path, line_num, "WHITE_ICON_ON_WHITE", line_clean))

print(f"Total potential issues found: {len(issues)}")
for path, line_no, issue_type, snippet in issues:
    print(f"[{path}:{line_no}] {issue_type}: {snippet[:130]}")

File: scratch/audit_all_elements.js
const fs = require('fs');
const path = require('path');

const rootDir = 'c:\\Users\\vc\\Documents\\nls-blog-hame\\nls-blog-hame';

function getAllHtmlFiles(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) {
            if (file !== '.git' && file !== 'node_modules' && file !== 'brain') {
                results = results.concat(getAllHtmlFiles(fullPath));
            }
        } else if (file.endsWith('.html')) {
            results.push(fullPath);
        }
    });
    return results;
}

const htmlFiles = getAllHtmlFiles(rootDir);
console.log(`Auditing ${htmlFiles.length} HTML files...`);

const report = [];

htmlFiles.forEach(filePath => {
    const relPath = path.relative(rootDir, filePath);
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');

    lines.forEach((line, idx) => {
        const lineNum = idx + 1;
        
        // 1. Look for text-white on light backgrounds
        if (line.includes('text-white') && !line.includes('dark:text-white')) {
            // Check if there is bg-white or light bg
            if (line.includes('bg-white') || line.includes('bg-slate-50') || line.includes('bg-slate-100') || line.includes('bg-gray-50') || line.includes('bg-surface')) {
                // If it doesn't have a solid colored class
                if (!line.includes('bg-sky-') && !line.includes('bg-blue-') && !line.includes('bg-primary') && 
                    !line.includes('bg-indigo-') && !line.includes('bg-emerald-') && !line.includes('bg-rose-') && 
                    !line.includes('bg-amber-') && !line.includes('bg-slate-800') && !line.includes('bg-slate-900') && 
                    !line.includes('bg-black') && !line.includes('bg-[#FF8A00]') && !line.includes('bg-[#0284c7]') && 
                    !line.includes('bg-[#131D38]') && !line.includes('from-') && !line.includes('bg-gradient')) {
                    report.push({ file: relPath, line: lineNum, type: 'TEXT_WHITE_ON_LIGHT', code: line.trim() });
                }
            }
        }

        // 2. Look for style="...color: white... or color: #fff on style="background: white...
        if ((line.includes('color: #fff') || line.includes('color: white')) && (line.includes('background: #fff') || line.includes('background: white') || line.includes('background-color: #fff'))) {
            report.push({ file: relPath, line: lineNum, type: 'INLINE_STYLE_WHITE_ON_WHITE', code: line.trim() });
        }

        // 3. Look for white logo image or icon on white bg
        if (line.includes('logo') && line.includes('white') && (line.includes('bg-white') || line.includes('bg-surface'))) {
            report.push({ file: relPath, line: lineNum, type: 'WHITE_LOGO_ON_LIGHT', code: line.trim() });
        }
    });
});

console.log(`Found ${report.length} potential issues:`);
report.forEach(r => {
    console.log(`[${r.file}:${r.line}] (${r.type}) ${r.code.substring(0, 140)}`);
});
